### DISCUSSION TITLE
Tutorial?

### DISCUSSION THREAD

**Discussion by Daniel15** [gircore/gir.core#1070] [General]
Is there a basic tutorial explaining the full end-to-end process of creating a basic Gtk app using this library? I'm looking at the example apps on the site (like Denaro), but their code is quite complex. I'd love to see a simple example app, like a todo list or something like that.

**Comment by badcel** [comment]
There are some minimal samples in the repository. The [use-page](https://gircore.github.io/docs/use.html ) links to it and should get you started.

There is some tutorial from @fsobolev [here](https://github.com/fsobolev/GirCoreExtrasTutorial).

A full fledged app will probably not land here as I don't want to keep it current myself.

I have some [personal project](https://github.com/badcel/Maus) to test things in the wild. It's a lot smaller than the big apps but did not see a release yet.

In general you can start with one of the minimal samples create some UI and use it like you want, there are no limits, use any other nuget you like.

For specific integration questions with linux you can check what the other apps are doing.

From my point of view the best practices should be aligned with the dotnet ones as you are "only" using a different UI framework.

If you care to improve the documentation in a way that would had helped you, feel free to create a pull request: the docs are part of this repo.
